src/modules/render/reservation.js

```diff
@@ -0,0 +1,192 @@
+  videoInstruction.className = 'reserve-details';
+  mealInfo.appendChild(videoInstruction);
+  const videoSpan = document.createElement('span');
+  videoSpan.className = 'video-instruction';
+  videoSpan.innerHTML = 'Video: ';
+  videoInstruction.appendChild(videoSpan);
+  const videoSrc = document.createElement('a');
+  videoSrc.href = 'meal.strYoutube';
+  videoSrc.innerHTML = 'YouTube Link';
+  videoInstruction.appendChild(videoSrc);
+
+  const sectionTitle = document.createElement('div');
+  reserveContent.appendChild(sectionTitle);
+  const sectionTitleHeader = document.createElement('div');
+  sectionTitleHeader.className = 'section-title';
+  const reservationCounterHeading = document.createElement('h2');
+  reservationCounterHeading.innerHTML = 'Reservations';
+  sectionTitleHeader.appendChild(reservationCounterHeading);
+  const reservationCounter = document.createElement('span');
+  reservationCounter.className = 'reservations-counter';
+  reservationCounterHeading.appendChild(reservationCounter);
+  sectionTitle.appendChild(sectionTitleHeader);
+  const reservationList = document.createElement('li');
+  reservationList.className = 'reservation-list';
+  API.getReservations(meal.idMeal).then((data) => {
+    if (data === 'No reservations available for this meal') {
+      reservationList.innerHTML = `<li class='no-comments'>${`${data}. Add a new reservation`}</li>`;
+    } else {
+      const reservationCounter = counter.reservation(data);
+      sectionTitleHeader.innerHTML = `Reservations(${reservationCounter})`;
+      data.forEach((userReservation) => {
+        displayReservations(reservationList, userReservation);
+      });
+    }
+  });
+  sectionTitle.appendChild(reservationList);
+
+  const reservationForm = document.createElement('div');
+  reservationForm.className = 'reservation-form';
+  reserveContent.appendChild(reservationForm);
+  const reservationFormTitle = document.createElement('h2');
+  reservationFormTitle.className = 'reservation-form-title';
+  reservationFormTitle.innerHTML = 'Add a reservation';
+  reservationForm.appendChild(reservationFormTitle);
+  const reserveForm = document.createElement('form');
+  reserveForm.className = 'form';
+  reservationForm.appendChild(reserveForm);
+  const inputReserveName = document.createElement('input');
+  inputReserveName.type = 'text';
+  inputReserveName.name = 'username';
+  inputReserveName.placeholder = 'Your Name';
+  inputReserveName.attributes.required = true;
+  reserveForm.appendChild(inputReserveName);
+  const inputReserveStartDate = document.createElement('input');
+  inputReserveStartDate.type = 'text';
+  inputReserveStartDate.name = 'start-date';
+  inputReserveStartDate.placeholder = 'Start Date (yyyy-mm-dd)';
+  inputReserveStartDate.attributes.required = true;
+  reserveForm.appendChild(inputReserveStartDate);
+  const inputReserveEndDate = document.createElement('input');
+  inputReserveEndDate.type = 'text';
```

Comment:
[OPTIONAL]
It would be great if the start and end date were of the type date instead of text. This will improve the UX of your reservation section on a larger scale thus giving users an easier time while inputting their date entries.

Reply:
Thank you @ZawadiSumba66 . We have worked on the video, the overflow content in the comments page has been edited and we've updated the date inputs to type: date